chain/sync.go

```diff
@@ -563,10 +563,6 @@ func (syncer *Syncer) Sync(ctx context.Context, maybeHead *types.TipSet) error {
 		)
 	}
 
-	if syncer.store.GetHeaviestTipSet().ParentWeight().GreaterThan(maybeHead.ParentWeight()) {
-		return nil
-	}
-
```

Comment:
This is already checked in `InformNewHead` and it depends on a value that is controlled by the sender so this extra check just adds a fake sense of security.